tasks/helpers/cleanup_stale_files.yml

```diff
@@ -0,0 +1,32 @@
+---
+# Sequence of tasks to detect and delete, if allowed, the files not
+# managed by the role anymore
+- name: List files
+  find:
+    paths: "{{ item.cleanup_directory }}"
+    patterns: "{{ item.pattern }}"
+  register: __cleanup_directory_files__
+- name: Check for stale files
+  set_fact:
+    __stale_files__: "{{
+      __cleanup_directory_files__.files | map(attribute='path') | map('basename') |
```

Comment:
It does not support nested files, right?

Reply:
Since the `find` is not recusrive by default yes,  `__cleanup_directory_files__` won't contain any nested directories contents.